File: jena/src/main/scala/pl/ostrzyciel/jelly/convert/jena/JenaConverterFactory.scala
package pl.ostrzyciel.jelly.convert.jena

import org.apache.jena.datatypes.RDFDatatype
import org.apache.jena.graph.{Node, Triple}
import org.apache.jena.sparql.core.Quad
import pl.ostrzyciel.jelly.core.proto.RdfStreamOptions
import pl.ostrzyciel.jelly.core.ConverterFactory

object JenaConverterFactory
  extends ConverterFactory[JenaProtoEncoder, JenaDecoderConverter, Node, RDFDatatype, Triple, Quad]:
  override protected def decoderConverter: JenaDecoderConverter = new JenaDecoderConverter()

  override final def encoder(options: RdfStreamOptions): JenaProtoEncoder = JenaProtoEncoder(options)

File: jena-java/src/main/java/eu/neverblink/jelly/convert/jena/JenaConverterFactory.java
package eu.neverblink.jelly.convert.jena;

import eu.neverblink.jelly.core.JellyConverterFactory;
import org.apache.jena.datatypes.RDFDatatype;
import org.apache.jena.graph.Node;

/**
 * A factory for creating Jena-specific encoder and decoder converters.
 * <p>
 * This class is a singleton and should be accessed via the {@link #getInstance()} method.
 */
public final class JenaConverterFactory
    extends JellyConverterFactory<Node, RDFDatatype, JenaEncoderConverter, JenaDecoderConverter> {

    private static final JenaConverterFactory INSTANCE = new JenaConverterFactory();

    private JenaConverterFactory() {}

    /**
     * Returns the singleton instance of the {@link JenaConverterFactory}.
     *
     * @return the singleton instance of {@link JenaConverterFactory}
     */
    public static JenaConverterFactory getInstance() {
        return INSTANCE;
    }

    @Override
    public JenaEncoderConverter encoderConverter() {
        return new JenaEncoderConverter();
    }

    @Override
    public JenaDecoderConverter decoderConverter() {
        return new JenaDecoderConverter();
    }
}
